### PR TITLE
Set scope of essential libs to provided

### PR DESCRIPTION
**Pull request by safris**
The opentracing-okhttp3 plugin instruments OkHttp3. If one wanted to use this plugin, he would use it in a runtime that has OkHttp3 (obviously, because if OkHttp3 is not in the runtime, then what's the point of the plugin). Therefore, it is guaranteed that the application runtime will have OkHttp3 on the classpath. The same argument applies to the OT API dependencies -- i.e. if tracing is desired, a tracer must be present, and thus the OT APIs will also be present in the runtime.

Currently, opentracing-okhttp3 has scope=compile dependencies for OkHttp3, which means that maven builds will always add the dependencies to the classpath of downstream modules that depend on opentracing-okhttp3 (an example of such a module is [SpecialAgent's OkHttp3 Agent Rule](https://github.com/opentracing-contrib/java-specialagent/blob/master/rules/opentracing-specialagent-okhttp/pom.xml#L40-L45). As can be seen, the POM is required to exclude the OkHttp3 libs. For SpecialAgent, this is necessary so as to avoid class loader issues with regard to multiple versions of the same classes being present in different class loaders.

### CODE REVIEW

File: opentracing-okhttp3/pom.xml [hunk 1]
```diff
@@ -13,10 +13,12 @@
     <dependency>
       <groupId>io.opentracing</groupId>
       <artifactId>opentracing-api</artifactId>
+      <scope>provided</scope>
```
Comment (pavolloffay) [inline opentracing-okhttp3/pom.xml]: I would say this should stay as in compile scope.

Comment (safris) [reply opentracing-okhttp3/pom.xml]: What is your reasoning for leaving opentracing-api as scope=compile?

Comment (pavolloffay) [reply opentracing-okhttp3/pom.xml]: if somebody includes this library it should compile. Making okhttp provided is fine that is expected to be found on the classpath.

Comment (safris) [reply opentracing-okhttp3/pom.xml]: Yes, I agree. I'll update the PR.

File: opentracing-okhttp3/pom.xml [hunk 2]
```diff
@@ -17,6 +17,7 @@
     <dependency>
       <groupId>io.opentracing</groupId>
       <artifactId>opentracing-noop</artifactId>
+      <scope>provided</scope>
```
Comment (pavolloffay) [inline opentracing-okhttp3/pom.xml]: Could you please rather remove noop? I am not sure why it is pulled in, I don't think it's used anywhere.

Comment (safris) [reply opentracing-okhttp3/pom.xml]: Removed!

**Review by pavolloffay — APPROVED** [review]

**Comment by pavolloffay** [comment]
merci @SevaSafris

**Comment by pavolloffay** [comment]
@SevaSafris do you need a release?

**Comment by safris** [comment]
Yes, that would be great. A release would allow me to close the related issue for SpecialAgent.

**Pull request merged**